src/CryptoNoteCore/CachedBlock.cpp

```diff
@@ -125,7 +125,7 @@ const BinaryArray& CachedBlock::getSignedBlockHashingBinaryArray() const {
     result.insert(result.end(), sig.begin(), sig.end());
   }
 
-  return blockHashingBinaryArray.get();
+  return signedBlockHashingBinaryArray.get();
```

Comment:
Consider using `blockHashingBinaryArray` and include signature into block's hash calculation starting from block version 5.